pkg/logs/tailers/file/rotate_nix.go

```diff
@@ -22,6 +22,9 @@ import (
 // - removed and recreated
 // - truncated
 func (t *Tailer) DidRotate() (bool, error) {
+	if t.osFile == nil {
```

Comment:
interesting in my source code I still have 
`	f, err := filesystem.OpenShared(t.fullpath)
	if err != nil {
		return false, fmt.Errorf("open %q: %w", t.osFile.Name(), err)
	}
	defer f.Close()
	offset := t.lastReadOffset.Load()

	st, err := f.Stat()
	if err != nil {
		return false, fmt.Errorf("stat %q: %w", f.Name(), err)
	}`
with that source code `t.fullpath` would be a safe option. But in new code I do not see it is used. Is t.fullpath is not correct usage anymore?

Reply:
I'm unsure altho `rotate_windows` still have it:
https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/tailers/file/rotate_windows.go#L22

Reply:
Ah, that explains it, I am an idiot.

So, what I was trying to say that if we call DidRotate AFTER `[tailer.setup](https://github.com/DataDog/datadog-agent/blob/e7000164ad5ba7b66a825b296b639d4b9afa9cad/pkg/logs/tailers/file/tailer_nix.go#L21-L25)` succeeded you can improve you change without changing condition and loosing file name by using t.fullpath.

Another word, instead of using 
```
    if t.osFile == nil {
		return false, fmt.Errorf("osFile is nil")
	}
	f, err := filesystem.OpenShared(t.osFile.Name())
	if err != nil {
		return false, fmt.Errorf("open %q: %w", t.osFile.Name(), err)
    }
```
you can return to original but safer form
```
	f, err := filesystem.OpenShared(t.fullpath)
	if err != nil {
		return false, fmt.Errorf("open %q: %w", t.fullpath, err)
    }
```
Which is cross-platform. Which is also would require change expression on Linux from filesystem.OpenShared(t.osFile.Name()) to filesystem.OpenShared(t.fullpath) which I am not sure if it will cause sideeffects.